src/commands/env/env-set.mjs

```diff
@@ -88,15 +89,29 @@ const setInMongo = async ({ api, key, siteInfo, value }) => {
 
 /**
  * Updates the env for a site configured with Envelope with a new key/value pair
- * @returns {Promise<object>}
+ * @returns {Promise<object | boolean>}
  */
-const setInEnvelope = async ({ api, context, key, scope, siteInfo, value }) => {
+const setInEnvelope = async ({ api, context, key, scope, secret, siteInfo, value }) => {
   const accountId = siteInfo.account_slug
   const siteId = siteInfo.id
+
+  // secret values must specify deploy contexts. `all` or `dev` are not allowed
+  if (secret && value && (!context || context.includes('dev'))) {
```

Comment:
I'm not entirely sure it's a great experience to have a different pattern for dev context but I understand we're trying to be clear about the expectations of secrets. In the future, if devs have an issue with this nuance we can always allow them to set dev with the secret flag (like they can do on the UI) it just won't be an actual secret (also true for those set on the dev context in UI)